### ISSUE TITLE
Wrong honorific used in FSD engaged script

### ISSUE THREAD

**Issue by richardbuckle** [EDCD/EDDI#734]
## EDDI version in which issue found
3.0.1-rc3

## Steps to reproduce

1. Jump from a non-imperial system to an imperial system.

## Expected
Output including "welcome back" with my Empire honorific.

## Observed
Was greeted with my Fed honorific:

>  Telemetry obtained. Convergence is in transit to the Gcirthi system. You have entered Empire space. Welcome back Post Captain. This is your first visit to this system. Gcirthi is an Empire Feudal with a population of around 23 and a half million people, allied with Yuri Grom. Gcirthi Gold United Group is the immediate faction. There are 3 orbital stations and a single planetary station in this system.

## Investigation
It appears to be an issue in the `FSD engaged` script. 

```
        {if lastsystem.allegiance && lastsystem.allegiance != "None" && lastsystem.allegiance != "Independent" && (!reportsystem.allegiance || reportsystem.allegiance = "None" || reportsystem.allegiance = "Independent"):
            You {OneOf("are leaving", "have left", "are no longer in")} {lastsystem.allegiance} space.
        |elif lastsystem.allegiance && lastsystem.allegiance != reportsystem.allegiance && system.allegiance && system.allegiance != "None":
            You {OneOf("are now in", "have entered", "are entering")} {reportsystem.allegiance} space.
            {if cmdr.title != "Commander":
                Welcome back {F("Honorific")}.
            }
        }
```

The `Honorific` is using the global `system.allegiance`. Consider replacing it with a Cottle function so that `reportsystem.allegiance` canb be passed in. Or update `system.allegiance` before calling `Honorific¬.

**Comment by Tkael** [comment]
I think I found a simple fix for this. PR #737 submitted.

**Comment by richardbuckle** [comment]
Fixed for 3.0.2-b1.
